community/kernel/src/main/java/org/neo4j/kernel/impl/api/OperationsFacade.java

```diff
@@ -439,6 +451,10 @@ public PrimitiveIntIterator relationshipGetPropertyKeys( long relationshipId ) t
         {
             return relationship.get().getPropertyKeys();
         }
+        finally
+        {
+            statement.assertOpen();
+        }
```

Comment:
If the iterator is lazy, then yes you are correct in that it wont protect against anything.
If it is not we have the same scenario as above. Right?